data-prepper-api/src/main/java/com/amazon/dataprepper/model/configuration/PluginModel.java

```diff
@@ -66,40 +113,73 @@ public PluginModelSerializer(final Class<PluginModel> valueClass) {
 
         public PluginModelDeserializer() {
-            this(null);
+            super(PluginModel.class, InternalJsonModel.class, PluginModel::new, InternalJsonModel::new);
         }
+    }
 
-        public PluginModelDeserializer(final Class<PluginModel> valueClass) {
+    /**
+     * Abstract deserializer for {@link PluginModel} objects. Any classes which inherit from {@link PluginModel}
+     * can extend this class. That class must also have a class derived from {@link InternalJsonModel}. It should be configured
+     * for Jackson databind. Please note that this class is only intended for internal use. Subclasses of {@link PluginModel}
+     * will need to create subclass of {@link AbstractPluginModelDeserializer}, but they do not need to override any methods.
+     * The subclass should only need to configure the correct classes in the default constructor.
+     *
+     * @param <T> The type inheriting from {@link PluginModel} that you ultimately need deserialized
+     * @param <M> The type inheriting from {@link InternalJsonModel} that has custom fields.
+     *
+     * @see SinkModel.SinkModelDeserializer
+     */
+    abstract static class AbstractPluginModelDeserializer<T extends PluginModel, M extends InternalJsonModel> extends StdDeserializer<PluginModel> {
+
+        private final Class<M> innerModelClass;
+        private final BiFunction<String, M, T> constructorFunction;
+        private final Supplier<M> emptyInnerModelConstructor;
+
+        protected AbstractPluginModelDeserializer(
+                final Class<T> valueClass,
+                final Class<M> innerModelClass,
+                final BiFunction<String, M, T> constructorFunction,
+                final Supplier<M> emptyInnerModelConstructor) {
             super(valueClass);
+            this.innerModelClass = innerModelClass;
+            this.constructorFunction = constructorFunction;
+            this.emptyInnerModelConstructor = emptyInnerModelConstructor;
         }
 
         @Override
         public PluginModel deserialize(final JsonParser jsonParser, final DeserializationContext context) throws IOException, JacksonException {
             final JsonNode node = jsonParser.getCodec().readTree(jsonParser);
 
             final Iterator<Map.Entry<String, JsonNode>> fields = node.fields();
-            Map.Entry<String, JsonNode> onlyField = fields.next();
+            final Map.Entry<String, JsonNode> onlyField = fields.next();
 
             final String pluginName = onlyField.getKey();
             final JsonNode value = onlyField.getValue();
 
-            final Map<String, Object> settingsMap = mapper.convertValue(value, new TypeReference<Map<String, Object>>(){});
-            return new PluginModel(pluginName, settingsMap);
+            M innerModel = SERIALIZER_OBJECT_MAPPER.convertValue(value, innerModelClass);
+            if(innerModel == null)
+                innerModel = emptyInnerModelConstructor.get();
```

Comment:
Take it or leave it:

Variable re-assignment can be problematic and hard to debug. I might recommend extracting these lines into a separate function to eliminate it.

```
private M convertValueOrGetDefault(final Object fromValue, final Class<T> toValueType, final Object defaultObject) {
    final M model = SERIALIZER_OBJECT_MAPPER.convertValue(value, innerModelClass);
    if(innerModel == null)
        return defaultObject;
    return model;
}
```
and then do the following assignment:
```
final M innerModel = convertValueOrGetDefault(value, innerModelClass, emptyInnerModelConstructor.get());